### PR TITLE
topk layer

### PR DESCRIPTION
**Pull request by LakshmiKumar23**
Layer implemented and model compiler changes

### CODE REVIEW

**Comment by kiritigowda** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- model_compiler/python/nnir.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/pullRequest?prid=5872559&bid=19220839)

**Comment by LakshmiKumar23** [comment]
@kiritigowda CI is failing because of Rali changes

File: amd_openvx_extensions/amd_nn/src/topk_layer.cpp [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+
+    ERROR_CHECK_STATUS(vxCopyTensorPatch((vx_tensor)parameters[5], 4, nullptr, nullptr, stride_output_0, values_ptr, VX_WRITE_ONLY, VX_MEMORY_TYPE_HOST));
+    ERROR_CHECK_STATUS(vxCopyTensorPatch((vx_tensor)parameters[6], 4, nullptr, nullptr, stride_output_1, indices_ptr, VX_WRITE_ONLY, VX_MEMORY_TYPE_HOST));
+
+    return VX_SUCCESS;
+}
+
+static vx_status VX_CALLBACK initializeTopK(vx_node node, const vx_reference *parameters, vx_uint32 num)
+{
+	data = new TopKLayerLocalData;
```
Comment (rrawther) [inline amd_openvx_extensions/amd_nn/src/topk_layer.cpp:194]: Adjust the spacing to be same for all the statemens

File: amd_openvx_extensions/amd_nn/src/topk_layer.cpp [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+	memset(data, 0, sizeof(*data));
+
+	vx_size input_dims_0[4];
+    vx_size count_input_dims_0 = input_dims_0[0]*input_dims_0[1]*input_dims_0[2]*input_dims_0[3];
+
+    ERROR_CHECK_STATUS(vxQueryTensor((vx_tensor)parameters[0], VX_TENSOR_DIMS, input_dims_0, sizeof(input_dims_0)));
+
+    data->x_tensor_buffer = (float*)malloc(count_input_dims_0*sizeof(float));
+
```
Comment (rrawther) [inline amd_openvx_extensions/amd_nn/src/topk_layer.cpp:203]: Looks like vxSetNodeAttribute(node, VX_NODE_LOCAL_DATA_PTR, ..) is not called for local data

File: amd_openvx_extensions/amd_nn/src/topk_layer.cpp [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+{
+	data = new TopKLayerLocalData;
+	memset(data, 0, sizeof(*data));
+
+	vx_size input_dims_0[4];
+    vx_size count_input_dims_0 = input_dims_0[0]*input_dims_0[1]*input_dims_0[2]*input_dims_0[3];
+
+    ERROR_CHECK_STATUS(vxQueryTensor((vx_tensor)parameters[0], VX_TENSOR_DIMS, input_dims_0, sizeof(input_dims_0)));
+
+    data->x_tensor_buffer = (float*)malloc(count_input_dims_0*sizeof(float));
```
Comment (rrawther) [inline amd_openvx_extensions/amd_nn/src/topk_layer.cpp:202]: May be you can combine this allocation with line #194 allocation so there is only one memory allocation

**Comment by kiritigowda** [comment]
@LakshmiKumar23 sync with TOT to pass the CI

File: amd_openvx_extensions/amd_nn/src/topk_layer.cpp (hunk 3 above)
Comment (LakshmiKumar23) [reply amd_openvx_extensions/amd_nn/src/topk_layer.cpp:202]: @rrawther AT line 194, I still don't know the size of the array. That's why couldn't initialize in the struct either

**Comment by LakshmiKumar23** [comment]
@kiritigowda synced with TOT and made changes mentioned by @rrawther

**Comment by kiritigowda** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- model_compiler/python/nnir.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/pullRequest?prid=5872559&bid=19220839)

**Comment by kiritigowda** [comment]
@LakshmiKumar23 is this ready to be merged?

**Comment by LakshmiKumar23** [comment]
@kiritigowda  Yes it's ready to be merged

**Pull request merged**